versionedRouter.js

```diff
@@ -871,6 +976,16 @@ router.post("/batch", ctx => {
 });
 
 const fileUpload = async ctx => {
+  const getReqMetadata = () => {
```

Comment:
Why do we need a callback here ? we already have `destType` in Ln 989, which can be used where `getReqMetadata` method is called right ?

Reply:
Where there is a need of firstEvent kind of thing, it probably makes sense to have a callback. But in methods like `handleProxyRequest`, `pollStatus`, `fileUpload` etc,. since the variable `destType` is directly available, it looks unnecessary.

Reply:
I made the implementation generic across all the endpoint handlers. All the metadata should be served from here.